pulsar-common/src/main/java/org/apache/pulsar/common/topics/TopicList.java

```diff
@@ -47,13 +47,16 @@ public static List<String> filterTopics(List<String> original, String regex) {
     }
     public static List<String> filterTopics(List<String> original, Pattern topicsPattern) {
 
-        final Pattern shortenedTopicsPattern = topicsPattern.toString().contains(SCHEME_SEPARATOR)
-                ? Pattern.compile(SCHEME_SEPARATOR_PATTERN.split(topicsPattern.toString())[1]) : topicsPattern;
+        final Pattern shortenedTopicsPattern = Pattern.compile(removeTopicDomainSchema(topicsPattern.toString()));
 
         return original.stream()
                 .map(TopicName::get)
+                .filter(topicName -> {
+                    String partitionedTopicName = topicName.getPartitionedTopicName();
+                    String removedSchema = SCHEME_SEPARATOR_PATTERN.split(partitionedTopicName)[1];
+                    return shortenedTopicsPattern.matcher(removedSchema).matches();
+                })
                 .map(TopicName::toString)
-                .filter(topic -> shortenedTopicsPattern.matcher(SCHEME_SEPARATOR_PATTERN.split(topic)[1]).matches())
                 .collect(Collectors.toList());
```

Comment:
It is best to deduplicate the results

Reply:
The result is a partition name(E.g. `public/default/tp-partition-0`), so we should not do deduplication.

Reply:
Ok, I see